sdk/Trace/BatchSpanProcessor.php

```diff
@@ -77,11 +82,23 @@ public function onStart(API\Span $span): void
      */
     public function onEnd(API\Span $span): void
     {
-        if (count($this->queue) < $this->maxQueueSize) {
-            $this->queue[] = $span;
+        if (null !== $this->exporter && $this->running) {
+            if (count($this->queue) < $this->maxQueueSize) {
+                $this->queue[] = $span;
+            }
+
+            if ($this->bufferReachedExportLimit() || $this->enoughTimeHasPassed()) {
+                $this->forceFlush();
+            }
         }
+    }
 
-        if ($this->bufferReachedExportLimit() && $this->enoughTimeHasPassed()) {
+    /**
+     * @inheritDoc
+     */
+    public function forceFlush(): void
```

Comment:
> ForceFlush()
Export all ended spans to the configured Exporter that have not yet been exported.
ForceFlush should only be called in cases where it is absolutely necessary, such as when using some FaaS providers that may suspend the process after an invocation, but before the Processor exports the completed spans.
ForceFlush should not block indefinitely. Language library authors can decide if they want to make the flush timeout configurable.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#forceflush